### ISSUE TITLE
Custom component with many localized fields freezes entire Builder editor and throws many errors in console

### ISSUE THREAD

**Issue by mimoid-prog** [BuilderIO/builder#3050]
**Describe the bug**
Custom component with many localized fields freezes entire Builder editor and throws many errors in console. The more localized fields the bigger frezee and lags occur.

**To Reproduce**
You can clone the repo from here: https://github.com/mimoid-prog/svelte-kit-builder-localized-error

1. Create custom component, define many fields with `localized: true`, register this component
2. Put component on the page
3. Try to edit it's properties
4. Entire Builder editor freezes, you can't click nothing, console is full of errors

**Expected behavior**
Properties should be available to edit instantly, there should not be errors in console.

**Screenshots**
Video: 

https://github.com/BuilderIO/builder/assets/55198084/c5c17535-bfdb-434b-b890-ea85b996aaf3

<img width="470" alt="Screenshot 2024-03-15 at 13 07 21" src="https://github.com/BuilderIO/builder/assets/55198084/7ac78815-796e-4936-bbdd-33f9f65c63eb">

**Additional context**
Node.js - 20.9.0

**Comment by samijaber** [comment]
@mimoid-prog We made a fix for the specific issue you shared a screenshot of.

However, another issue persists. It is specifically tied to the Content Input `favoriteTechStack` being a localized list. We are actively investigating this and will follow up with you.

I want to point out that you are setting `localized: true` to both the list and all of its internal individual fields. That is unusual, as you would usually either want to localize the entire array, or the individual objects within, but not both. 

If you remove the `localized` key from the parent list:

```diff
{
			name: 'favoriteTechStack',
			type: 'list',
			defaultValue: [
				{
					name: 'Svelte',
					description: 'A compiler for building user interfaces.'
				}
			],
-			localized: true,
			subFields: [
				{
					name: 'name',
					type: 'text',
					defaultValue: 'Svelte',
					localized: true
				},
				{
					name: 'description',
					type: 'text',
					defaultValue: 'A compiler for building user interfaces.',
					localized: true
				}
			]
		}
```

Then your example now works and no longer freezes. Would that be satisfactory in unblocking you? If not: what need do you have for localizing the entire array?

**Comment by samijaber** [comment]
Closing this issue given its staleness and lack of followup. Feel free to reopen if the issue persists
